src/awkward/_connect/julia/utils.py

```diff
@@ -0,0 +1,257 @@
+# BSD 3-Clause License; see https://github.com/scikit-hep/awkward-1.0/blob/main/LICENSE
+
+import os
+import warnings
+from pathlib import Path
+
+from julia.api import JuliaError
+
+from awkward import __version__
+
+# FIXME: move this out?
+__awkward_array_jl_version__ = "1.0.0-DEV"
+
+juliainfo = None
+julia_initialized = False
+julia_kwargs_at_initialization = None
+julia_activated_env = None
+
+
+def _load_juliainfo():
+    """Execute julia.core.JuliaInfo.load(), and store as juliainfo."""
+    global juliainfo
+
+    if juliainfo is None:
+        from julia.core import JuliaInfo
+
+        try:
+            juliainfo = JuliaInfo.load(julia="julia")
+        except FileNotFoundError:
+            env_path = os.environ["PATH"]
+            raise FileNotFoundError(
+                f"Julia is not installed in your PATH. Please install Julia and add it to your PATH.\n\nCurrent PATH: {env_path}",
+            ) from None
+
+    return juliainfo
+
+
+def is_julia_version_greater_eq(juliainfo=None, version=(1, 9, 0)):
+    """Check if Julia version is greater than specified version."""
+    if juliainfo is None:
+        juliainfo = _load_juliainfo()
+    current_version = (
+        juliainfo.version_major,
+        juliainfo.version_minor,
+        juliainfo.version_patch,
+    )
+    return current_version >= version
+
+
+def _check_for_conflicting_libraries():  # pragma: no cover
```

Comment:
I'm not sure if we want to check this. I've run into some inconsistencies between `PythonCall` and `PyCall`. It could be just my environment: the installation order or outdated versions.

Reply:
I think we should be happy to enforce a lower bound; with our "special" dependencies we can be stricter, and that makes our lives easier. I've simplified this a bit, so now 1.9.0 is definitely a lower bound.